File: problems/beginner/1005/1005.rs
use std::io;

fn get_number_from_stdin() -> f64 {
	let mut num = String::new();

	io::stdin()
		.read_line(&mut num)
		.expect("Error while reading from stdin");

	let num: f64 = num
		.trim()
		.parse()
		.expect("Error while parsing string into number");

	num
}

fn main() {
	const FIRST_WEIGHT: f64 = 3.5; 
	const SECOND_WEIGHT: f64 = 7.5; 
	const TOTAL_WEIGHT: f64 = 11 as f64; 

	let a = get_number_from_stdin();
	let b = get_number_from_stdin();

	let result: f64 = ((a * FIRST_WEIGHT) + (b * SECOND_WEIGHT)) / TOTAL_WEIGHT;

	println!("MEDIA = {:.5}", result);
}

File: problems/beginner/1005/1005.dart
import 'dart:convert';
import 'dart:io';

String input() {
  return (stdin.readLineSync(encoding: utf8) as String);
}

double doubleInput() {
  return double.parse(input().trim());
}

double weightedAverage(double a, double b) {
  const FIRST_WEIGHT = 3.5;
  const SECOND_WEIGHT = 7.5;
  const TOTAL_WEIGHT = FIRST_WEIGHT + SECOND_WEIGHT;

  return (a * FIRST_WEIGHT + b * SECOND_WEIGHT) / TOTAL_WEIGHT;
}

void main() {
  var a = doubleInput();
  var b = doubleInput();

  print('MEDIA = ${weightedAverage(a, b).toStringAsFixed(5)}');
}
